python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -61,46 +86,142 @@ def make_dagster_job_from_airflow_dag(
         tags (Dict[str, Field]): Job tags. Optionally include
             `tags={'airflow_execution_date': utc_date_string}` to specify execution_date used within
             execution of Airflow Operators.
-        use_airflow_template_context (bool): If True, will call get_template_context() on the
-            Airflow TaskInstance model which requires and modifies the DagRun table. The use_airflow_template_context
-            setting is ignored if use_ephemeral_airflow_db is True.
-            (default: False)
         unique_id (int): If not None, this id will be postpended to generated op names. Used by
             framework authors to enforce unique op names within a repo.
-        mock_xcom (bool): If True, dagster will mock out all calls made to xcom, features that
-            depend on xcom may not work as expected. (default: False)
-        use_ephemeral_airflow_db (bool): If True, dagster will create an ephemeral sqlite airflow
-            database for each run. (default: False)
         connections (List[Connection]): List of Airflow Connections to be created in the Ephemeral
             Airflow DB, if use_emphemeral_airflow_db is False this will be ignored.
 
     Returns:
         JobDefinition: The generated Dagster job
 
     """
-    pipeline_def = make_dagster_pipeline_from_airflow_dag(
-        dag=dag,
+    check.inst_param(dag, "dag", DAG)
+    tags = check.opt_dict_param(tags, "tags")
+    unique_id = check.opt_int_param(unique_id, "unique_id")
+    connections = check.opt_list_param(connections, "connections", of_type=Connection)
+
+    connections = check.opt_list_param(connections, "connections", of_type=Connection)
+
+    if IS_AIRFLOW_INGEST_PIPELINE_STR not in tags:
+        tags[IS_AIRFLOW_INGEST_PIPELINE_STR] = "true"
+
+    tags = validate_tags(tags)
+
+    node_dependencies, node_defs = get_graph_definition_args(dag=dag, unique_id=unique_id)
+
+    serialized_connections = serialize_connections(connections)
+
+    @resource(
+        config_schema={
+            "dag_location": Field(str, default_value=dag.fileloc),
+            "dag_id": Field(str, default_value=dag.dag_id),
+            "connections": Field(
+                Array(inner_type=dict),
+                default_value=serialized_connections,
+                is_required=False,
+            ),
+        }
+    )
```

Comment:
Is this likely to be configured at runtime/in a a sensor without a code push? If not you might consider making this a vanilla object